src/Link.cc

```diff
@@ -440,6 +440,39 @@ void Link::AddWorldForce(EntityComponentManager &_ecm,
   this->AddWorldWrench(_ecm, _force, torque);
 }
 
+//////////////////////////////////////////////////
+void Link::AddLinkForce(EntityComponentManager &_ecm,
+                        const math::Vector3d &_force) const
+{
+  auto worldPose = _ecm.ComponentData<components::WorldPose>(this->dataPtr->id)
+                       .value_or(sim::worldPose(this->dataPtr->id, _ecm));
+
+  // The force is expressed in terms of the link coordinates, but
+  // ExternalWorldForcecmd applies the force expressed in world coordinates
+  // so we need to compute the force expressed in world coordinates
+  math::Vector3d worldForce = worldPose.Rot() * _force;
+
+  // Apply Force using AddWorldForce method
+  this->AddWorldForce(_ecm, worldForce);
+}
+
+//////////////////////////////////////////////////
+void Link::AddLinkForce(EntityComponentManager &_ecm,
```

Comment:
Can you add a unit test for this?
https://github.com/gazebosim/gz-sim/blob/gz-sim9/src/Link_TEST.cc